pkg/reconciler/instances/eventing/preaction/removenatsoperatorstep.go

```diff
@@ -0,0 +1,99 @@
+package preaction
+
+import (
+	"strings"
+
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/chart"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/instances/eventing/log"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/kubernetes/kubeclient"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	removeNatsOperatorStepName = "removeNatsOperator"
+	natsOperatorDeploymentName = "nats-operator"
+	natsOperatorLastVersion    = "1.24.7"
+	natsSubChartPath           = "eventing/charts/nats"
+	eventingNats               = "eventing-nats"
+	oldConfigValue             = "global.image.repository"
+	newConfigValue             = "eu.gcr.io/kyma-project"
+)
+
+var (
+	natsOperatorCRDsToDelete = []string{"natsclusters.nats.io", "natsserviceroles.nats.io"}
+)
+
+type kubeClientProvider func(context *service.ActionContext, logger *zap.SugaredLogger) (*kubeclient.KubeClient, error)
+
+type removeNatsOperatorStep struct {
+	kubeClientProvider
+}
+
+func newRemoveNatsOperatorStep() *removeNatsOperatorStep {
+	return &removeNatsOperatorStep{
+		kubeClientProvider: defaultKubeClientProvider,
+	}
+}
+
+func defaultKubeClientProvider(context *service.ActionContext, logger *zap.SugaredLogger) (*kubeclient.KubeClient, error) {
+	kubeClient, err := kubeclient.NewKubeClient(context.Task.Kubeconfig, logger)
```

Comment:
[You don't need this fct.](https://github.com/kyma-incubator/reconciler/blob/main/pkg/reconciler/service/action.go#L15). The `kubeclient.KubeClient` won't become accessible soon ([PR is open for this change](https://github.com/kyma-incubator/reconciler/pull/384/files#diff-7db9da48a12f7d364dd6a81c34f9d6d1fa160e2bd9ced27e0778e5b78ec67f1dR3))